src/components/BottomSheet/BottomSheet.tsx

```diff
@@ -0,0 +1,212 @@
+    // if (keyboardHeight) {
+
+    // }
+    // return -keyboard.height.value / animatedPosition.value
+    return Math.abs(0)
+  }, [])
+  const animatedBottomSheetState = useDerivedValue(() => {
+    if (animatedPositionIndex.value === -1) {
+      return BOTTOM_SHEET_STATE.close
+    } else {
+      return BOTTOM_SHEET_STATE.open
+    }
+  }, [])
+  const animatedContentHeight = useDerivedValue(() => {
+    if (animatedBottomSheetState.value === BOTTOM_SHEET_STATE.close) {
+      animatedContainerTranslateY.value = 0
+      return withTiming(animatedPosition.value)
+    }
+    return animatedContainerTranslateY.value - animatedKeyboardHeight.value
+  }, [])
+
+  // const animatedSnapPoints = useNormalizedSnapPoints(
+  //   _providedSnapPoints,
+  //   animatedContainerHeight,
+  //   // animatedContentHeight,
+  // )
+
+  const animatedStyles = useAnimatedStyle(
+    () => ({
+      transform: [{translateY: animatedContentHeight.value}],
+      height: animatedPosition.value,
+    }),
+    [],
+  )
+
+  // callbacks
+  const handlePositionIndex = useCallback(
+    (newIndex: number) => {
+      animatedPositionIndex.value = newIndex
+    },
+    [animatedPositionIndex],
+  )
+  const handleOpen = useCallback(() => {
+    handlePositionIndex(0)
+    setPointerEvents('auto')
+  }, [handlePositionIndex])
+  const handleClose = useCallback(() => {
+    handlePositionIndex(-1)
+    setPointerEvents('none')
+  }, [handlePositionIndex])
+
+  const pan = Gesture.Pan()
+    .onBegin(() => {})
+    .onChange(event => {
+      animatedContainerTranslateY.value = Math.max(0, event.translationY)
+    })
+    .onFinalize(() => {
+      if (animatedContainerTranslateY.value >= animatedPosition.value / 2) {
+        runOnJS(handleClose)()
+      } else {
+        animatedContainerTranslateY.value = withTiming(0)
+      }
+    })
+
+  // styles
+  const contentStyles = useMemo(
+    () => [{backgroundColor: 'white'}, _providedStyle, animatedStyles],
+    [_providedStyle, animatedStyles],
+  )
+
+  // context
+  const contextValues = useMemo(
+    () => ({
+      animatedContainerTranslateY,
+      shouldHandleKeyboardEvents,
+      animatedPositionIndex,
+      animatedPosition,
+
+      // callbacks
+      open: handleOpen,
+      close: handleClose,
+    }),
+    [
+      animatedContainerTranslateY,
+      animatedPosition,
+      animatedPositionIndex,
+      handleClose,
+      handleOpen,
+      shouldHandleKeyboardEvents,
+    ],
+  )
+
+  useImperativeHandle(ref, () => contextValues)
+
+  return (
+    <BottomSheetProvider value={contextValues}>
+      <GestureHandlerRootView style={StyleSheet.absoluteFillObject} pointerEvents={pointerEvents}>
+        <BottomSheetBackdropContainer animatedPositionIndex={animatedPositionIndex} />
+        <BottomSheetContainer
+        // containerHeight={_providedContainerHeight}
+        // containerOffset={animatedContainerOffset}
+        // topInset={topInset}
+        // bottomInset={bottomInset}
```

Comment:
please refactor to remove those unused comment

Reply:
this component isn't final and that comment is helpful for the next update